### DISCUSSION TITLE
Trying to recreate AFDB's visualization of pLDDT NewCartoon colors

### DISCUSSION THREAD

**Discussion by rbdavid** [BradyAJohnston/MolecularNodes#391] [Q&A] [answered]
Hi, maybe this is a basic question. I am trying to recreate the stepped colors used in AlphaFold Database's structural visualization of predicted models. Specifically, the structures have the residues' pLDDT values stored in the b-factor column of the .pdb file. My logic for setting up the node tree follows the solution for Q&A #269 where multiple color_set/selection node pairs are used, one for each of the steps in pLDDT color scheme (see below). 

![image](https://github.com/BradyAJohnston/MolecularNodes/assets/8484541/a46adf3e-b571-4e4e-b0e3-2bef9c87b49e)

Unfortunately, the MN_select_attribute selection node does not have the b-factor values as a possible attribute to select by. 

Alternatively, there is the MN_color_attribute_map coloring node but this only allows for up to a three color map with various color interpolation styles. 

I'm fairly new to blender and geometry nodes, so what's a good way of implementing this qualitative color scheme?

**Comment by rbdavid** [comment]
Looking at the inner mechanism of the MN_select_attribute, maybe I can create a boolean return value based on residues' b-factor attributes (within the various ranges of the color scheme) and use this to select atoms to then be colored with the respective color from the scheme?

**Comment by rbdavid** [comment]
Group input fields for the desired atom selection node will be:
- Attribute (expect a float)
- Min (used as a condition input for a Math Compare node)
- Max (used as a condition input for a Math Compare node)

Group output from the selection node will be an Atoms group that passed the boolean tests. 

After some digging through the default selection node groups, I think this is going to mirror the MN_select_res_id_range node group but use the b_factor named attribute and Float.

**Comment by BradyAJohnston** [comment, marked as answer]
Hi mate, thanks for asking the question! You could do the different coloring based on boolean selections as in #269, but the cleaner way would be to have a couple of switch nodes as in the example below: 

![CleanShot 2024-01-03 at 07 35 46@2x](https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/388b0d87-bb0b-4f77-8702-ebff267f8de3)

![CleanShot 2024-01-03 at 07 44 17@2x](https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/d54a4d64-162c-4ca1-a840-5003a1846196)

This group of nodes can become a node group with inputs (for the cutoffs) and outputs (the color), which you can create by selecting them and creating a group with <kbd>Cmd/Ctrl</kbd> + <kbd>G</kbd>. This is useful node group and one that I will include in the future release of MN as I'm sure there will be lots of AlphaFold visualisations going on with it.

**Reply by rbdavid** [reply]
FYI, the last boolean test should be greater than 90.

**Reply by BradyAJohnston** [reply]
ah yep, whoops!

**Comment by rbdavid** [comment]
I stepped away for a bit but here's where I got. 

![Screenshot (39)](https://github.com/BradyAJohnston/MolecularNodes/assets/8484541/81347307-51d5-498b-b28c-312496165245)

where the MN_select_res_by_attribute node group is 

![Screenshot (40)](https://github.com/BradyAJohnston/MolecularNodes/assets/8484541/46a394ec-a844-414b-8220-363fcd111688)

Just for any other interested parties: the hex codes for the AFDB colors are FF7D45, FFDB13, 65CBF3, and 0053D6.
